src/SprykerEco/Zed/CrefoPayApi/Business/Logger/CrefoPayApiLogger.php

```diff
@@ -0,0 +1,170 @@
+     */
+    protected $entityManager;
+
+    /**
+     * @param \SprykerEco\Zed\CrefoPayApi\Persistence\CrefoPayApiEntityManagerInterface $entityManager
+     */
+    public function __construct(CrefoPayApiEntityManagerInterface $entityManager)
+    {
+        $this->entityManager = $entityManager;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\CrefoPayApiRequestTransfer $requestTransfer
+     * @param \Generated\Shared\Transfer\CrefoPayApiResponseTransfer $responseTransfer
+     * @param string $requestType
+     *
+     * @return \Generated\Shared\Transfer\PaymentCrefoPayApiLogTransfer
+     */
+    public function logApiCall(
+        CrefoPayApiRequestTransfer $requestTransfer,
+        CrefoPayApiResponseTransfer $responseTransfer,
+        string $requestType
+    ): PaymentCrefoPayApiLogTransfer {
+        $paymentCrefoPayApiLog = $this->createPaymentCrefoPayApiLogTransfer(
+            $requestTransfer,
+            $responseTransfer,
+            $requestType
+        );
+
+        return $this->getTransactionHandler()->handleTransaction(function () use ($paymentCrefoPayApiLog) {
+            return $this->entityManager->savePaymentCrefoPayApiLog($paymentCrefoPayApiLog);
+        });
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\CrefoPayApiRequestTransfer $requestTransfer
+     * @param \Generated\Shared\Transfer\CrefoPayApiResponseTransfer $responseTransfer
+     * @param string $requestType
+     *
+     * @return \Generated\Shared\Transfer\PaymentCrefoPayApiLogTransfer
+     */
+    protected function createPaymentCrefoPayApiLogTransfer(
+        CrefoPayApiRequestTransfer $requestTransfer,
+        CrefoPayApiResponseTransfer $responseTransfer,
+        string $requestType
+    ): PaymentCrefoPayApiLogTransfer {
+        return (new PaymentCrefoPayApiLogTransfer())
+            ->setRequestType($requestType)
+            ->setCrefoPayOrderId($this->getCrefoPayOrderId($requestTransfer, $requestType))
+            ->setIsSuccess($responseTransfer->getIsSuccess())
+            ->setResultCode($this->getResultCode($responseTransfer, $requestType))
+            ->setMessage($this->getMessage($responseTransfer, $requestType))
+            ->setSalt($this->getSalt($responseTransfer, $requestType))
+            ->setRequest($requestTransfer->serialize())
+            ->setResponse($responseTransfer->serialize());
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\CrefoPayApiRequestTransfer $requestTransfer
+     * @param string $requestType
+     *
+     * @throws \SprykerEco\Zed\CrefoPayApi\Business\Exception\InvalidRequestTypeException
+     *
+     * @return string
+     */
+    protected function getCrefoPayOrderId(CrefoPayApiRequestTransfer $requestTransfer, string $requestType): string
+    {
+        $method = sprintf(static::GET_REQUEST_METHOD, ucfirst($requestType));
+
+        if (!method_exists($requestTransfer, $method)) {
+            throw new InvalidRequestTypeException();
```

Comment:
It would be great if you can add some message to this exception in order to get developer some information.